### PR TITLE
Lower the minSdkVersion to 9.

### PR DESCRIPTION
**Pull request by thagikura**
Changes include:
- Use methods from the ViewCompat otherwise those methods were not supported in
  < 14
- Use compat version of Fragment and Preferences for the demo app

The ./gradlew build runs the lint check to verify the methods that aren't
supported in the API level < 14 are not called.
So it should be checked by the Circle CI.

### CODE REVIEW

**Comment by thagikura** [comment]
@yaraki Could you have a look at this PR to double check?

File: app/build.gradle [hunk 1]
```diff
@@ -39,7 +39,10 @@ android {
 
 dependencies {
     compile project(":flexbox")
+    compile "com.android.support:support-v4:${rootProject.ext.supportLibVersion}"
     compile "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
+    compile "com.android.support:preference-v7:${rootProject.ext.supportLibVersion}"
+    compile "com.android.support:preference-v14:${rootProject.ext.supportLibVersion}"
```
Comment (yaraki) [inline app/build.gradle]: Do you need this?

Comment (thagikura) [inline app/build.gradle]: This is a nice to have to keep the material theme on API level 14+.

See http://stackoverflow.com/questions/32070670/preferencefragmentcompat-requires-preferencetheme-to-be-set

Added explanation to the values-v14/styles.xml
https://github.com/google/flexbox-layout/pull/45/files#diff-b7d8affb5986661f0b964437076193dbR20

Comment (thagikura) [inline app/build.gradle]: Friendly ping? @yaraki

**Comment by yaraki** [comment]
+2

**Comment by thagikura** [comment]
Thanks. Once I merge this CL, the demo app won't be built unless the new version of the library is released.

I'd like to wait for #34 to be included in the same release if that PR doesn't take so long.

**Comment by thagikura** [comment]
Ah the previous comment was not correct... The demo app will compile after this PR is merged (the demo app have a reference to the module (source code) dependency instead of the library dependency)

Merging this PR.

**Pull request merged**